src/mockserver.js

```diff
@@ -4,91 +4,121 @@ import { logger } from '@user-office-software/duo-logger';
+  const respondToPostRequest = function (request) {
+    if (request.method !== 'POST') {
+      return;
+    }
+    let responsePath;
+    const requestXml = String(request.body.xml);
+    const match = requestXml.match('<tns:(.*?)>');
+    const method = match[1];
 
-        if (
-            method === 'getBasicPersonDetailsFromUserNumber' ||
-            method === 'getSearchableBasicPeopleDetailsFromUserNumbers'
-        ) {
-            let regexp = '';
-            method === 'getSearchableBasicPeopleDetailsFromUserNumbers'
-                ? (regexp = '<UserNumbers>(.*?)<')
-                : (regexp = '<UserNumber>(.*?)<');
-            const match = requestXml.match(regexp);
+    if (
+      method === 'getBasicPersonDetailsFromUserNumber' ||
+      method === 'getSearchableBasicPeopleDetailsFromUserNumbers'
+    ) {
+      let regexp = '';
+      method === 'getSearchableBasicPeopleDetailsFromUserNumbers'
+        ? (regexp = '<UserNumbers>(.*?)<')
+        : (regexp = '<UserNumber>(.*?)<');
+      const match = requestXml.match(regexp);
 
-            responsePath = 'src/responses/user/' + method + '/' + match[1] + '.xml'
-        }
-        if (method === 'getSearchableBasicPersonDetailsFromEmail') {
-            const match = requestXml.match('<Email>(.*?)<');
+      responsePath = 'src/responses/user/' + method + '/' + match[1] + '.xml';
+    }
+    if (method === 'getSearchableBasicPersonDetailsFromEmail') {
+      const match = requestXml.match('<Email>(.*?)<');
 
-            responsePath = 'src/responses/user/' + method + '/' + match[1] + '.xml'
-        }
-        if (
-            method === 'getBasicPeopleDetailsFromUserNumbers' &&
-            requestXml.includes('<UserNumbers>')
-        ) {
-            responsePath = 'src/responses/user/notEmptyResponse' + '.xml'
-        }
-        if (responsePath === null || responsePath === undefined) {
-            responsePath = 'src/responses/user/' + method + '.xml'
-        }
-        const file = fs.readFileSync(responsePath, 'utf8')
-        return {
-            body: file,
-        };
+      responsePath = 'src/responses/user/' + method + '/' + match[1] + '.xml';
+    }
+
+    if (
+      method === 'getPersonDetailsFromSessionId' &&
+      requestXml.includes('<SessionId>')
+    ) {
+      const match = requestXml.match('<SessionId>(.*?)<');
+      if (match[1].toString() === 'externalUser') {
+        responsePath = 'src/responses/user/' + method + '/4.xml';
+      } else {
+        responsePath = 'src/responses/user/' + method + '/1.xml';
+      }
+    }
+
+    if (method === 'getRolesForUser') {
+      const match = requestXml.match('<userNumber>(.*?)<');
+
+      responsePath = 'src/responses/user/' + method + '/' + match[1] + '.xml';
+    }
+    if (
+      method === 'getBasicPeopleDetailsFromUserNumbers' &&
+      requestXml.includes('<UserNumbers>')
+    ) {
+      const match = requestXml.match('<UserNumbers>(.*?)<');
+      if (match[1] === 1 || match[1] === 4) {
```

Comment:
It would be useful to have a comment explaining magic numbers 1 and 4 here. These are the user numbers of the default local users from the seeded database right?

Reply:
Was taking it from cypress support initialDBData.ts .But will add a comment for it.